smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/PublisherDecorator.java

```diff
@@ -0,0 +1,20 @@
+package io.smallrye.reactive.messaging;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+
+/**
+ * SPI to allow extension of publishers included in the final graph
+ */
+public interface PublisherDecorator {
```

Comment:
Should this be in an SPI package?

Is this class intended for possible spec inclusion in the future? If so, it should probably be in the API artifact instead

Reply:
I'm not sure, I wasn't really thinking about it being used outside of this project. Would you like it moved to an SPI project?

I don't think there are any plans to put something like this in the specification. If it turns out to be useful for more than just the metrics integration then we could add it, though it would be good to have a few more use cases so we know if there's more information that needs to be provided to the decorator.

Reply:
I wasn't sure if this could be something that's standardized at some point, even if only for how metrics are handled with Publishers.

Not wedded to any particular approach, was curious as to future thoughts